detox/src/utils/Deferred.test.js

```diff
@@ -0,0 +1,75 @@
+const Deferred = require('./Deferred');
+
+describe('Deferred', () => {
+  /** @type {Deferred} */
+  let deferred;
+
+  describe('static properties', () => {
+    it('should have "pending" status', () =>
+      expect(Deferred.PENDING).toBe('pending'));
+
+    it('should have "resolved" status', () =>
+      expect(Deferred.RESOLVED).toBe('resolved'));
+
+    it('should have "rejected" status', () =>
+      expect(Deferred.REJECTED).toBe('rejected'));
+  });
```

Comment:
Why is this needed?

Reply:
Why not? Does it break some principles of good design or something? I just prefer to explicitly test existence of those three static properties, to not mix that concern into other tests.